pkg/kp/rc.go

```diff
@@ -0,0 +1,30 @@
+package kp
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/rc"
+)
+
+// ReplicationControllerStore represents an interface for persisting replication controllers to Consul,
+// as well as restoring replication controllers from Consul.
+type ReplicationControllerStore interface {
+	// Create creates a replication controller with the specified manifest and selectors.
+	// The node selector is used to determine what nodes the replication controller may schedule on.
+	// The pod label set is:
+	// 1) applied to every pod the replication controller schedules,
+	// 2) used (in addition to replication controller ID) to determine what pods are already scheduled.
+	Create(manifest pods.Manifest, nodeSelector labels.Selector, podLabels labels.Set) (ReplicationControllerHandle, error)
+
+	// List lists all of the replication controllers.
+	List() ([]ReplicationControllerHandle, error)
```

Comment:
and should there be a method that queries for a *specific* replication controller, taking in some ID?